heat/core/dndarray.py

```diff
@@ -74,6 +79,74 @@ def device(self):
     def dtype(self):
         return self.__dtype
 
+    @property
+    def halo(self):
+        """
+        Returns
+        -------
+        int : number of rows or columns used as halo
+        """
+        return self.__halo
+
+    @halo.setter
+    def halo(self, halo):
+        """
+        Parameters
+        ----------
+        halo : None or int
+            Number of rows or columns to use as halo along the split axis
+        """
+        # check if tensor is distributed
+        if not self.is_distributed():
+            warnings.warn("halos are only supported when tensor is distributed")
+            return
+
+        halo = self.sanitize_halo(halo)
+        has_halo = self.is_halorized()
+
+        # if tensor is already halorized ...
+        if has_halo:
+            # ... check if new halo differs from current one
+            if halo == self.__halo:
+                # new halo size equals current halo, so no need to do anything
+                warnings.warn(
+                    "provided halo is equal to current halo, no further actions taken",
+                    RuntimeWarning,
+                )
+                return
+            else:
+                # new halo size is different from current one, so delete current halos
+                # TODO: is there a special way to delete torch tensor?
```

Comment:
`
del tensor
`

Reply:
Okay thanks, i wasn't sure this was sufficient.